### ISSUE TITLE
Transform组件使用的问题

### ISSUE THREAD

**Issue by tomcatter** [Unity-Technologies/com.unity.uiwidgets#276]
```
new Transform(
                transform: transform,
                alignment: FractionalOffset.center,
                child: new Column(
                    mainAxisAlignment: Unity.UIWidgets.rendering.MainAxisAlignment.center,
                    crossAxisAlignment: Unity.UIWidgets.rendering.CrossAxisAlignment.center,
                    children: new List<Widget>
                    {
                         inviteImage
                           ...
                    }
                    )
                );
```

`transfrom`变量，初始化如下：
```
          var transform = new Matrix4(
                     1, 0, 0, 0,
                     0, 1, 0, 0,
                     0, 0, 1, 0,
                     0, 0, 0, 1
                     );

            transform.setIdentity();
            transform.setEntry(3, 2, 0.01f);
            //transform.rotateX(0.01f * _offset.dy);
            transform.rotateY(0.01f * -15);
```

其中`inviteImage`是我自己封装的`MouseRegion` 在hover的时候改变背景图片，代码如下：

```
new MouseRegion(
                child: new Container(
                    child: this._isHover ? hoverImage : normalImage
                    ),
                    onEnter: (e) =>
                    {
                        setState(() =>
                        {
                            this._isHover = true;
                        });
                        this.widget.onPressed?.Invoke();
                    },
                    onExit: (e) =>
                    {
                        setState(() =>
                        {
                            this._isHover = false;
                        });
                    }
                );
```

现在的问题是： 根据我上面设置的`transform`参数，应该显示如下图: 

![211214183330](https://user-images.githubusercontent.com/7018011/145981825-8635c5fa-28fc-4cc8-976a-de7e7ee551b7.png)

有`Unity Editor`中运行后，鼠标点击`Game`窗口，然后发现，这个控件的`transform`不起作用了（没有了上面的倾斜效果，如下图），但是鼠标移出`Game`窗口后又恢复正常，鼠标移出移入`Game`窗口，控件就在这两种状态下变化，`transform`的属性没有更改。

![0211214183619](https://user-images.githubusercontent.com/7018011/145982254-639ed3e2-574f-4c76-971e-6b4ef628fc3c.png)

上面描述的是在`MouseRegion`中现象。

然后我又尝试了不使用`MouseRegion`，也即
```
new Container( child: normalImage )
```
直接返回一张图片，这时不再发生鼠标移入移出后效果改变的现象

**Comment by zhuxingwei** [comment]
Hi Thanks for the report!

We will look into this issue later today and let you know when it is fixed or further more information is required. Thanks !

**Comment by zhuxingwei** [comment]
Hi we have tested your sample with flutter and it turns out that **this is a bug in flutter** (in version 1.17.5 at least).
The test code in dart is as follows:
```
class TestTransformHover extends StatefulWidget {
  @override
  TestTransformHoverState createState() {
    return TestTransformHoverState();
  }
}

class TestTransformHoverState extends State< TestTransformHover> {

  @override
  Widget build(BuildContext context) {
     var transform = Matrix4(
        1, 0, 0, 0,
        0, 1, 0, 0,
        0, 0, 1, 0,
        0, 0, 0, 1
    );

     transform.setIdentity();
     transform.setEntry(3, 2, 0.01);
     transform.rotateY(0.01 * -15);

     var home = Transform(
         transform: transform,
         alignment: FractionalOffset.center,
         child: new InviteImage()
     );
     return home;
  }
}

class InviteImage extends StatefulWidget
{
  @override
  InviteImageState createState() {
    return InviteImageState();
  }
}

class InviteImageState extends State<InviteImage>
{
  bool isHover = false;

  @override
  Widget build(BuildContext context) {
    return  MouseRegion(
        child:  Container(child: Center(
        child:  Container(width: 100, height: 100, color: isHover ? Colors.blue: Colors.green))
    ),
    onEnter: (evt)  {
          setState(() {
            isHover = true;
          });
    },
    onExit: (evt) {
          setState(() {
            isHover = false;
          });
    },
    );
  }
}
```
![1639732767910](https://user-images.githubusercontent.com/4134652/146520508-5071d520-ed2e-465f-9101-2ff5cc18e63c.jpg)

The transform looks to be properly rendered. However, we also find that since there is no _mouse_  in a mobile device, the result is misleading: The code that generates a active `MouseHoverRegion` for the `MouseRegion` is not executed at all: 
![1639732786157](https://user-images.githubusercontent.com/4134652/146520552-37344e59-2981-45af-a747-4c25fcef733c.jpg)

After modifying this line in flutter by adding the contents in the blue box which force the `MouseHoverRegion` to accept _mouseHover_ events, the render result is shown below: we can see that this is exactly the issue that you found.
![1639732792998](https://user-images.githubusercontent.com/4134652/146520571-790ca801-f2d3-44cc-84b1-e0cf01829d6a.jpg)

We will continue to investigate this issue further more and will let you know when we have any new findings.

**Comment by zhuxingwei** [comment]
This issue is solved by #287

**Comment by tomcatter** [comment]
经验证，问题解决
